crates/bevy_app/src/app.rs

```diff
@@ -115,6 +118,7 @@ impl App {
         for sub_app in self.sub_apps.values_mut() {
             (sub_app.runner)(&mut self.world, &mut sub_app.app);
         }
+        self.stage_very_last.run(&mut self.world);
```

Comment:
Maybe we should consider bumping time management as a concern into the "app runner"? It seems like what we really want is "very first" (as in pre-input event processing for a given update), not "very last".

Conceptually "starting the stopwatch at the beginning of a frame" feels easier to reason about and ensures we don't have multiple frames with zero elapsed time.

Although I'm not convinced winit will make that switch easy on us. Processing input after deciding to present seems desirable, but they abstract that out from us.

Reply:
I'm not sure if we want "very first". Ideally we'd be setting the time as close to when the frame is presented as possible. Which means during the render stage. For example, there might be a future where people are inserting systems after the frame is presented like `bevy_framepace` is doing. (*This positioning doesn't actual work for frame pacing in pipelined rendering. It actually does want to be before input processing. But it's late and I'm having trouble coming up with something that would actually be inserted after frame presentation.*) This hypothetical system could cause the same timing issues that we're currently seeing with input processing. 

I did consider adding the time update to after app.update in the event handler, but any other runners would need to know to add the call there as well.

When we do have pipelined rendering I suspect that we'll want to run the time update system right after the frame presentation in the render stage and then move the time into the app world during extract. This won't currently work very well because that would cause the time to be for the N-2 frame. But in pipelined rendering it would be for the N-1 frame, where N is the current frame. Ideally ideally we'd be getting the frame time from hardware, but that requires api support that isn't fully supported on all hardware/os's/wgpu yet.

Reply:
I suppose we can move towards something more like the future with pipelines rendering by adding another extract stage at the end of the rendering schedule. This stage would logically be merged with the other one once we have pipelined rendering.

Reply:
>When we do have pipelined rendering I suspect that we'll want to run the time update system right after the frame presentation in the render stage and then move the time into the app world during extract. 

This would also be my suggestion based on my experimenting with `bevy_framepace`. However, instead of using extract we could try using a globally shared `Arc<RwLock>`, or use a channel.

Reply:
> However, instead of using extract we could try using a globally shared Arc<RwLock>, or use a channel.

A channel sounds like a good idea. I'll try that.